aten/src/ATen/native/TensorShape.cpp

```diff
@@ -550,6 +550,26 @@ std::vector<Tensor> tensor_split(const Tensor& self, IntArrayRef indices, int64_
   return splits;
 }
 
+std::vector<Tensor> tensor_split(const Tensor& self, const Tensor& tensor_indices_or_sections, int64_t dim) {
+  TORCH_CHECK(self.dim() > 0, "expected at least a 1-dimensional tensor");
+  auto split_device = tensor_indices_or_sections.device();
+  TORCH_CHECK(split_device == kCPU,
+    "tensor_split expected tensor_indices_or_sections to be on cpu, but its on ", split_device);
+  auto split_dtype = tensor_indices_or_sections.scalar_type();
+  TORCH_CHECK(split_dtype == at::kLong,
+  "tensor_split expected tensor_indices_or_sections to have dtype of long, got ", split_dtype);
+  auto split_dim = tensor_indices_or_sections.dim();
+  if (split_dim == 0) {
+    int64_t sections = tensor_indices_or_sections.item<int64_t>();
+    return self.tensor_split(sections, dim);
+  } else {
+    TORCH_CHECK(split_dim == 1, "tensor_split expected indices to be a 1-dimensional tensor");
```

Comment:
This check should still be moved up or refined, because tensor_split expects `tensor_indices_or_sections` to be either a zero dimensional or one dimensional tensor. There's no user-facing concept of "indices" that the message can reference.

If the check is moved outside this if statement, for example, it would very that tensor_indices_or_sections has zero or 1 dimensions.